Client/src/Unified/Services/Submitter/BatchUntilInactiveBlock.cs

```diff
@@ -0,0 +1,249 @@
+// This file is part of the ArmoniK project
+//
+// Copyright (C) ANEO, 2021-$CURRENT_YEAR$. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+//   J. Fonseca        <jfonseca@aneo.fr>
+//   D. Brasseur       <dbrasseur@aneo.fr>
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+
+using JetBrains.Annotations;
+
+namespace ArmoniK.DevelopmentKit.Client.Unified.Services.Submitter;
+
+/// <summary>
+///   Provides a dataFlow block that batches inputs into arrays.
+///   A batch is produced when the number of currently queued items becomes equal
+///   to BatchSize, or when a Timeout period has elapsed after receiving the last item.
+/// </summary>
+public class BatchUntilInactiveBlock<T> : IPropagatorBlock<T, T[]>, IReceivableSourceBlock<T[]>
```

Comment:
You seem to forward many methods to your member `source_`. Maybe your class _is a_ `BatchBlock<T>`.

Reply:
partialy true. It's a custom Batchblock<T> to add trigger timer option inside. To create one a custom implementation of IPropagator is requested